plugins/quetz_sql_authenticator/tests/test_quetz_sql_authenticator.py

```diff
@@ -0,0 +1,105 @@
+from contextlib import contextmanager
+from unittest import mock
+
+from click.testing import CliRunner
+from quetz_sql_authenticator.cli import _create, _delete, _reset, calculate_hash
+from quetz_sql_authenticator.db_models import Credentials
+
+
+# Test invalid login
+def test_invalid_login(client):
+    response = client.post(
+        "/auth/sql/authorize",
+        data={"username": "testuser", "password": "testpassword"},
+    )
+    assert response.status_code == 200
+    assert "login failed" in response.text
+
+
+# Test valid login
+def test_valid_login(client, db):
+    # Insert user
+    credentials = Credentials(
+        username="testuser", password=calculate_hash("testpassword")
+    )
+    db.add(credentials)
+
+    response = client.post(
+        "/auth/sql/authorize",
+        data={"username": "testuser", "password": "testpassword"},
+    )
+    # Assert that we get a redirect to the main page
+    assert response.status_code == 303
+
+
+# Test create user
+def test_create_user(db):
+    @contextmanager
+    def get_db():
+        yield db
```

Comment:
what's the idea behind this context manager?

Reply:
The idea is to mock the `get_db_manager` context manager, I took that from the [conda suggest tests](https://github.com/janjagusch/quetz/blob/58f40741c87a080d8463a2dbda384375215bc7a1/plugins/quetz_conda_suggest/tests/test_quetz_conda_suggest.py#L33)